radix-engine/benches/wasm_benchmarks.rs

```diff
@@ -0,0 +1,311 @@
+    let import_object = wasmer::imports! {};
+
+    // instantiate
+    wasmer::Instance::new(&module, &import_object).expect("Failed to instantiate module")
+}
+
+fn get_wasmi_call_host_instance(
+    engine: &wasmi::Engine,
+    mut store: wasmi::StoreContextMut<HostState>,
+    code: &[u8],
+    func: fn(i64, i64, i64) -> i64,
+) -> wasmi::Instance {
+    let module = wasmi::Module::new(&engine, code).unwrap();
+    let mut linker = <wasmi::Linker<HostState>>::new();
+    let host_func = wasmi::Func::wrap(
+        &mut store,
+        move |_caller: wasmi::Caller<'_, HostState>,
+              x: i64,
+              y: i64,
+              cnt: i64|
+              -> Result<i64, wasmi::core::Trap> {
+            let x = func(x, y, cnt);
+            Ok(x)
+        },
+    );
+    linker.define("host", "host_func", host_func).unwrap();
+
+    linker
+        .instantiate(store.as_context_mut(), &module)
+        .unwrap()
+        .ensure_no_start(store.as_context_mut())
+        .unwrap()
+}
+
+macro_rules! bench_ops {
+    ($t:literal, $ops:literal, $x:expr, $y:expr, $range:expr) => {
+        paste::item! {
+            pub fn [< $t _ $ops _benchmark >] (c: &mut Criterion) {
+                let func_name = concat!($t, "_", $ops);
+                let mut group = c.benchmark_group(func_name);
+                let wasm_code = get_wasm_file();
+                let wasm_call_host_code = wabt::wat2wasm(&WAT_CALL_HOST).unwrap();
+
+                // wasmi stuff
+                let engine = wasmi::Engine::default();
+                let mut store = wasmi::Store::new(&engine, 42);
+                let wasmi_instance = get_wasmi_instance(&engine, store.as_context_mut(), &wasm_code[..]);
+                let wasmi_func = wasmi_instance
+                    .get_typed_func::<(i64, i64, i64), i64>(store.as_context_mut(), func_name)
+                    .unwrap();
+
+                // wasmi stuff that calls native method
+                let mut store_call_host = wasmi::Store::new(&engine, 42);
+                let wasmi_call_host_instance = get_wasmi_call_host_instance(&engine, store_call_host.as_context_mut(), &wasm_call_host_code[..], [< $t _ $ops >]);
+                let wasmi_call_host_func = wasmi_call_host_instance
+                    .get_typed_func::<(i64, i64, i64), i64>(store_call_host.as_context_mut(), "local_call_host_func")
+                    .unwrap();
+
+                // wasmer stuff
+                let wasmer_instance = get_wasmer_instance(&wasm_code[..]);
+                let wasmer_func = wasmer_instance.exports.get_function(func_name).unwrap();
+
+                for i in $range {
+                    // native
+                    group.bench_with_input(BenchmarkId::new("rust-native", i), &i, |b, i| {
+                        black_box(b.iter(|| {
+                            for _ in 0..*i {
+                                [< $t _ $ops >] ($x, $y, 0);
+                            }
```

Comment:
Should we move the `black_box` to the statement inside the `for` loop, so compiler won't strip the code completely?